server/core/src/main/java/org/infinispan/server/core/backup/resources/CacheResource.java

```diff
@@ -235,11 +236,7 @@ public CacheBackupEntry next() {
                cmd.setInternalMetadata(entry.internalMetadata);
                return cmd;
             })
-            .flatMap(cmd -> RxJavaInterop.voidCompletionStageToFlowable(invocationHelper.invokeAsync(cmd, 1)));
-      // The count will subscribe.
-      // Execute the complete publisher as blocking, this will avoid (hide, more precisely) issues when operations
-      // are blocking by mistake.
-      return Flowable.fromPublisher(blockingManager.blockingPublisher(p))
+            .flatMap(cmd -> RxJavaInterop.voidCompletionStageToFlowable(invocationHelper.invokeAsync(cmd, 1)), batchSize)
```

Comment:
Technically this flatMap may not even be doing any parallelism if the invokeAsync completes synchronous. This may be why the blockingPublisher was moved to where it is, to provide some parallelism. But we should really be observing on a non blocking thread instead in the flatMap if we want to guarantee that. Or even better would be to use the parallel operator as in the example here:

```java
return Flowable.fromPublisher(blockingManager.blockingPublisher(f))
            .parallel(batchSize)
               .runOn(nonblockingManager.asScheduler())
               .map(entry -> {
                  ...
               })
               // Technically map and concatMap can be combined
               .concatMap(cmd -> RxJavaInterop.voidCompletionStageToFlowable(invocationHelper.invokeAsync(cmd, 1)))
            .sequential()
            .count()
            .toCompletionStage()
            .thenAccept(entries -> log.debugf("Cache %s restored %d entries", cacheName, entries));
```

Unfortunately, it requires more refactoring to get the nonBlockingManager reference, so I have not done that in this PR.